web/src/connectors/task-annotator-connector/utils.ts

```diff
@@ -0,0 +1,129 @@
+import { CategoryDataAttrType } from '../../api/typings';
+import { PageSize } from '../../shared/components/document-pages/document-pages';
+
+interface Token {
+    id: number;
+    text: string;
+    x: number;
+    y: number;
+    width: number;
+    height: number;
+}
+
+interface Data {
+    tokens: Token[];
+    dataAttributes: CategoryDataAttrType[];
+}
+
+interface PageInfoObj {
+    id: number;
+    type: string;
+    bbox: number[];
+    category: string;
+    data: Data;
+    children: number[];
+    text: string;
+}
+
+interface Page {
+    page_num: number;
+    size: PageSize;
+    objs: PageInfoObj[];
+    revision: string;
+    user_id: string;
+    pipeline: number;
+    date: string;
+    is_validated: boolean;
+    categories: string[];
+}
+
+export interface LatestRevisionObj {
+    [page_num: string]: Page[];
+}
+
+export interface RevisionObjByUser {
+    [user_id: string]: {
+        [page_num: number]: {
+            size: PageSize;
+            objs: PageInfoObj[];
+        };
+    };
+}
+
+export interface UserRevision {
+    user_id: string;
+    page_num: number;
+    size: PageSize;
+    objs: PageInfoObj[];
+    revision: string;
+    isValidated: boolean;
+    date: string;
+    pipeline: number;
+    categories: string[];
+}
+
+export function convertToRevisionByUserArray(sourceObj: LatestRevisionObj): UserRevision[] {
```

Comment:
About naming, it's better to avoid names like `somethingObj`/`somethingArr`/`somethingString`

variable name should provide useful information for those who will read this code, `Obj` doesn't give any useful information, so it's better to make up new meaningful name or at least remove `Obj` from its name

Reply:
https://github.com/epam/badgerdoc/pull/584/commits/f9b5691c7ba4bc23a4908ca62b8610d799576f54